kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/internal/data/TokenBucket.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.core.internal.data;
+
+public class TokenBucket {
+
+    private final int capacity;
+    // period in mills between 1 token refill
+    private final long refillPeriod;
+    private int remainingTokens;
+    private long lastRefillTime;
+
+    public TokenBucket(int capacity, long refillPeriod) {
+        this.capacity = capacity;
+        this.remainingTokens = capacity;
+        this.refillPeriod = refillPeriod;
+        this.lastRefillTime = System.currentTimeMillis();
+    }
+
+    public boolean getToken() {
+        boolean result = false;
+        refill();
+        if (isTokenAvailable()) {
+            this.remainingTokens--;
+            result = true;
+        }
+        return result;
+    }
+
+    private boolean isTokenAvailable() {
+        return this.remainingTokens != 0;
+    }
+
+    private void refill() {
+        long now = System.currentTimeMillis();
+        if (now - this.lastRefillTime >= this.refillPeriod) {
+            this.remainingTokens = (int) Math.min(this.capacity,
+                    this.remainingTokens + (now - this.lastRefillTime) / this.refillPeriod);
+            this.remainingTokens = Math.max(1, this.remainingTokens);
+            this.lastRefillTime = now;
+        }
+    }
+
+    public long getNextRefillEta() {
```

Comment:
Maybe add a comment that you don't get an updated and meaningful value until getToken() is called. This usage is misleading and error prone however.
The method name should also indicate a "time to next refill" (relative to now) instead of an "ETA of the next refill" (relative to epoch).